plugin.video.newson/default.py

```diff
@@ -0,0 +1,270 @@
+    def openURL(self, url):
+        try:
+            cacheResponce = self.cache.get(ADDON_NAME + '.openURL, url = %s'%url)
+            if not cacheResponce:
+                request = urllib2.Request(url)
+                request.add_header('User-Agent','Mozilla/5.0 (Windows; U; MSIE 9.0; Windows NT 9.0; en-US)')
+                page = urllib2.urlopen(request, timeout = TIMEOUT)
+                log(page.headers['content-type'])
+                results = json.load(page)
+                page.close()
+                self.cache.set(ADDON_NAME + '.openURL, url = %s'%url, results, expiration=datetime.timedelta(hours=1))
+            return self.cache.get(ADDON_NAME + '.openURL, url = %s'%url)
+        except urllib2.URLError, e:
+            log("openURL Failed! " + str(e), xbmc.LOGERROR)
+        except socket.timeout, e:
+            log("openURL Failed! " + str(e), xbmc.LOGERROR)
+        except:            
+            xbmcgui.Dialog().notification(ADDON_NAME, 'Site unreachable, try again later...', ICON, 4000)
+            
+        
+    def mainMenu(self):
+        log('mainMenu')
+        for item in MENU:
+            self.addDir(*item)
+        
+                    
+    def browseMenu(self, id=1):
+        log('browseMenu, id = ' + str(id))
+        self.stateMenu = [tuple(s.format(id) for s in tup) for tup in self.stateMenu]
+        for item in self.stateMenu:
+            self.addDir(*item)
+            
+            
+    def getStates(self):
+        log('getStates')
+        state     = []
+        stateLST  = []
+        data = self.openURL(BASE_API)
+        if not data:
+            return []
+        for channel in data:
+            state.append(channel['config']['state'])
+        states = collections.Counter(state)
+        for key, value in sorted(states.iteritems()):
+            stateLST.append(("%s"%(key), key , '{}'))
+        del state[:]
+        return stateLST
+            
+            
+    def newsCasts(self, state):
+        log('newsCasts, state = ' + state)
+        urls = []
+        data = self.openURL(BASE_API)
+        if not data:
+            return
+        for channel in data:
+            if state in channel['config']['state']:
+                chid   = channel['identifier']
+                title  = channel['title']
+                icon   = (channel['icon'] or ICON)
+                for idx, stream in enumerate(channel['streams']):
+                    #multiple urls, only and unique.
+                    url = (stream['Url'].split('.m3u8?')[0])+'.m3u8'
+                    offset = stream['OffsetFromNow']
+                    delay  = url+'&delay=%d'
+                    #todo do something with delay option?
+                    if url not in urls:
+                        urls.append(url)
+                        chid = chid+'.%d'%idx if idx > 0 else chid
+                        label      = "%s - %s" % (chid, title)
+                        infoLabels ={"label":label ,"title":label}
```

Comment:
ic there is another mediatype, was a bit confused. 
https://forum.kodi.tv/showthread.php?tid=250936&pid=2319280#pid2319280